content-src/components/Search/Search.jsx

```diff
@@ -32,64 +31,36 @@ export class _Search extends React.PureComponent {
 
   doSearchHandoff(text) {
     this.props.dispatch(ac.OnlyToMain({type: at.HANDOFF_SEARCH_TO_AWESOMEBAR, data: {text}}));
+    this.props.dispatch({type: at.FAKE_FOCUS_SEARCH});
     this.props.dispatch(ac.UserEvent({event: "SEARCH_HANDOFF"}));
     if (text) {
-      // We don't hide the in-content search if there is no text (user hit <Enter>)
       this.props.dispatch({type: at.HIDE_SEARCH});
     }
   }
 
   onSearchHandoffClick(event) {
     // When search hand-off is enabled, we render a big button that is styled to
-    // look like a search textbox. If the button is clicked with the mouse, we
-    // focus it. If the user types, transfer focus to awesomebar.
-    // If the button is clicked from the keyboard, we focus the awesomebar normally.
-    // This is to minimize confusion with users navigating with the keyboard and
-    // users using assistive technologoy.
+    // look like a search textbox. If the button is clicked, we style
+    // the button as if it was a focused search box and show a fake cursor but
+    // really focus the awesomebar without the focus styles ("hidden focus").
```

Comment:
I would copy most or part of this comment in the Reducer as well to explain fakeFocus.